### DISCUSSION TITLE
Supabase policy results in empty array after executing RPC function

### DISCUSSION THREAD

**Discussion by kncfreight** [supabase/supabase#16818] [Questions]
In order to activate RLS for my project, I am using the following Supabase policy for my table "quotations":

(customername = ((auth.jwt() -> 'app_metadata'::text) ->> 'customername'::text))
when I hardcode the policy to the needed value it works fine:
(customername = 'testCompany')

the app_metadata.customername is filled with the correct value string and has the data type "text". 
Other rpc functions work well with this policy as well.

the rpc function takes in an object as argument:

{
  p_customername: 'testCompany',
  p_pod: 'Santos [BRSSZ]',
  p_zipcodes: null,
  p_incoterm: 'exw',
  p_typ: 'Export',
  p_modus: 'FCL',
  p_transportmode: 'Oceanfreight',
  p_carrier: null,
  p_startrow: 0,
  p_endrow: 50
}
BEGIN
    RETURN QUERY
        SELECT *
        FROM quotations
        WHERE (quotations.customername = p_customername)
        AND (p_pod is null or quotations.pod = p_pod)
        AND (p_zipcodes is null or quotations.transportplace = p_zipcodes)
        AND (p_incoterm is null or quotations.incoterm = p_incoterm)
        AND (p_typ is null or quotations.typ = p_typ)
        AND (p_modus is null or quotations.modus = p_modus)
        AND (p_transportmode is null or quotations.transportmode = p_transportmode)
        AND (p_carrier is null or quotations.carrier = p_carrier)
        ORDER BY id DESC
        OFFSET p_startrow
        LIMIT p_endrow - p_startrow + 1;
END;

when I turn of RLS, the rpc function works fine. if I turn it on, there is no error, but the result is simply an empty array.

I really don't get it. Can you kindly help out? Thank you!

**Comment by GaryAustin1** [comment]
That looks like a plpgsql function.   You can do `raise log 'appmeta = %',auth.jwt()->app_metadata;` and see if the jwt is getting to the server correctly.  The result will be in the postgres logs of the dashboard.
